### PR TITLE
Refactoring of VBCommandLineParser

### PR DESCRIPTION
**Pull request by AdamSpeight2008**
By introducing a mutable class `Mutable_VBCommandLineArguments`, it has made it possible to extract some of the code inside the `Parse` method out into their own functions. 
Creation of the final immutable instance of `VisualBasicCommandLineArguments` is now a function of the Immutable.

### CODE REVIEW

File: src/Compilers/VisualBasic/Portable/CommandLine/Mutable_VBCommandLineArguments.vb [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+
+    Friend Class Mutable_VBCommandLineArguments
+
+        Protected Friend Sub New(args As IEnumerable(Of String), c As VisualBasicCommandLineParser, baseDirectory As String, IsScriptRunner As Boolean)
+            Me.IsScriptRunner = IsScriptRunner
+            scriptArgs = If(IsScriptRunner, New List(Of String)(), Nothing)
+            c.FlattenArgs(args, diagnostics, flattenedArgs, scriptArgs, baseDirectory, responsePaths)
+            Me.BaseDirectory = baseDirectory
+            outputDirectory = Me.BaseDirectory
+            Me.c = c
```
Comment (AdamSpeight2008) [inline src/Compilers/VisualBasic/Portable/CommandLine/Mutable_VBCommandLineArguments.vb]: I thought I had removed this in last commit.

**Comment by davkean** [comment]
tag @dotnet/roslyn-compiler

**Comment by davkean** [comment]
Adam, you now have [4 outstanding PRs](https://github.com/dotnet/roslyn/pulls/Adamspeight2008) against the VB compiler, some of which now becoming stale. 

To reduce the distraction/context switching of the compiler team which have to review the changes, I'd like you drive these to conclusion before submitting anymore PRs.

**Comment by gafter** [comment]
This change is too large to be seen in a PR.

Is there an issue suggesting this refactoring where the proposed work was discussed? I'd like to see such a discussion.

**Comment by AdamSpeight2008** [comment]
@gafter Just added an issue #7205. The change does restructure and alter that particular file a bit.

**Comment by tmat** [comment]
:-1: 
Although the ```Parse``` method is long it is straightforward and easy to understand. I don't see why would it need to be changed.

**Comment by jaredpar** [comment]
Bringing in my feedback from the issue:

> I still don't understand the justification for making this change. Yes the code is monolithic but the structure is well known to anyone who has done command line parsing. I don't see any advantage in breaking it up into multiple functions here.

:-1:

**Pull request closed without merging**